### PR TITLE
[core] Add lumped mass vector method to element

### PR DESCRIPTION
**Pull request by riccardotosi**
**Description**
Following #7934, we add a new method to compute the lumped mass vector to the base element.

Please mark the PR with appropriate tags: 
- Cpp, Feature

**Changelog**
Please summarize the changes in one list to generate the changelog:
- Added new method `CalculateLumpedMassVector`
- Expose `CalculateLumpedMassVector` to Python

### CODE REVIEW

**Comment by loumalouomega** [comment]
@riccardotosi you should override the method in the solid element

**Review by rubenzorrilla — CHANGES_REQUESTED** [review]
Blocking to prevent accidental merge. The behaviour of the base class should be discussed.

**Comment by loumalouomega** [comment]
@riccardotosi if you do blind PR be careful (I know that I have done some :P)

**Comment by philbucher** [comment]
TBD:
Due to the usage of this fct I think it should be const

**Comment by loumalouomega** [comment]
The previous one in the base solid element was const...

**Comment by rubenzorrilla** [comment]
@KratosMultiphysics/technical-committee agrees on the method addition. However:
- The base class method must throw an error if called. Note that this enforces all the elements that aim to use the new explicit strategies to implement it (@miguelmaso , @riccardotosi , @KratosMultiphysics/fluid-dynamics)
- The method must be `const`

Please add the new additions in a follow up PR.

**Comment by riccardotosi** [comment]
> 
> 
> @KratosMultiphysics/technical-committee agrees on the method addition. However:
> 
>     * The base class method must throw an error if called. Note that this enforces all the elements that aim to use the new explicit strategies to implement it (@miguelmaso , @riccardotosi , @KratosMultiphysics/fluid-dynamics)
> 
>     * The method must be `const`
> 
> 
> Please add the new additions in a follow up PR.

Ok, I will do it.

As an aside comment, any reason for opening another PR and not using this one?

**Comment by rubenzorrilla** [comment]
> > @KratosMultiphysics/technical-committee agrees on the method addition. However:
> > ```
> > * The base class method must throw an error if called. Note that this enforces all the elements that aim to use the new explicit strategies to implement it (@miguelmaso , @riccardotosi , @KratosMultiphysics/fluid-dynamics)
> > 
> > * The method must be `const`
> > ```
> > 
> > 
> > Please add the new additions in a follow up PR.
> 
> Ok, I will do it.
> 
> As an aside comment, why opening another PR and not using this one?

Because we want the KratosCore base element addition (and the required changes to make it compile as the structural mechanics ones in this PR) to be separated from all the new additions in other applications.

File: kratos/python/add_mesh_to_python.cpp [hunk 1]
```diff
@@ -469,6 +475,7 @@ void  AddMeshToPython(pybind11::module& m)
     .def("Calculate", &ElementCalculateInterface<Matrix >)
     .def("CalculateMassMatrix", &ElementCalculateMassMatrix)
     .def("CalculateDampingMatrix", &ElementCalculateDampingMatrix)
+    .def("CalculateLumpedMassVector", &ElementCalculateLumpedMassVector)
```
Comment (philbucher) [inline kratos/python/add_mesh_to_python.cpp:478]: this is minor, but do we need this to be exposed?

Comment (riccardotosi) [reply kratos/python/add_mesh_to_python.cpp:478]: I am not sure. I added for sake of consistency.

Comment (rubenzorrilla) [reply kratos/python/add_mesh_to_python.cpp:478]: I was thinking the same. But considering that the other elemental contributions are exposed, I'd keep them for the sake of consistency (they might eventually be useful for debugging or for more academic purposes).

**Comment by riccardotosi** [comment]
> 
> 
> @KratosMultiphysics/technical-committee agrees on the method addition. However:
> 
>     * The base class method must throw an error if called. Note that this enforces all the elements that aim to use the new explicit strategies to implement it (@miguelmaso , @riccardotosi , @KratosMultiphysics/fluid-dynamics)
> 
>     * The method must be `const`
> 
> 
> Please add the new additions in a follow up PR.

I wrongly interpreted this message. After talking privately with @rubenzorrilla, I have applied all core changes in this PR. This way, #7947 will contain only applications changes.

**Review by rubenzorrilla — APPROVED** [review]
I'm fine if CI passes.

**Review by philbucher — APPROVED** [review]
:+1: 

@KlausBSautter FYI
Once this is in we could also take a look at the other elements in StructuralMechanics

**Pull request merged**